script.js

```diff
@@ -1,12 +1,602 @@
+  </div>
+  `;
+  headerElem.innerHTML += searchFieldsHtml ;
+
+  let pageSetupHtml = `
+    <div id="allShowsContainer" class="container-fluid show-content">
+      <div id="freeSearchResult" class="row" style="display: none">
+        <h2 class="row shows-genre-headers">Search Result</h2>
+        <div id="freeSearchResultShows" class="card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+        </div>
+      </div>
+      <h2 class="row shows-genre-headers">Top Rated Shows</h2>
+      <div id="topRatedShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Drama Shows</h2>
+      <div id="dramaShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Comedy Shows</h2>
+      <div id="comedyShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Crime Shows</h2>
+      <div id="crimeShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Action Shows</h2>
+      <div id="actionShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+      <h2 class="row shows-genre-headers">Sci-Fi Shows</h2>
+      <div id="scifiShows" class="shows-genre-card-group card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4">
+      </div>
+    </div>
+    <div id="showContainer" class="container">
+      <div id="castListSection">
+        <h2 class="row cast-list-header">Cast</h2>
+        <div id="castList" class="card-group row flex-nowrap row-cols-1 row-cols-md-2 row-cols-lg-3 g-4" style='display:none'>
+        </div>
+      </div>
+      <h2 class="row episodes-header">Episodes</h2>
+      
+      <div id="seasonInputContainer" class="row" style="display:none">
+        <div class="col-md-4 offset-md-1">
+          <input id='input' type="text" class="input form-control" placeholder="search episodes">
+        </div>
+        
+        <div class="col-md-3">
+          <select class="form-select" id="seasonSelect">
+          </select>
+        </div>
+        <div class="col-md-3">
+          <select id='selectEpisode' name="" class="form-select">
+            <option value="">All Episodes</option>
+          </select>
+        </div>
+        <p id="epAmount" class="col-10 offset-md-1"></p>
+      </div>
+      <div id="seasonEpisodes" class="row show-content">
+      </div>
+    </div>
+  `;
+  rootElem.innerHTML = pageSetupHtml;
+
+})();
+
+function makeCastList(castList){
+  let castListDiv = document.getElementById('castList');
+  castListDiv.style.display = 'flex';
```

Comment:
Does this need to be done in JS? Could you simply apply a style to this ID in your CSS?

Reply:
its set to none when on the main/homepage and is to be displayed only when on the show/episodes page. its triggered on and off by different functions as well (when logo pressed goes back to home page requiring the display to be set to none). I can look into it further to see if I can do it in a better way without it being done I JS.